src/cpp/session/modules/SessionThemes.cpp

```diff
@@ -42,10 +46,52 @@ namespace themes {
 
 namespace {
 
+const std::string kDefaultThemeLocation = "/theme/default/";
+const std::string kGlobalCustomThemeLocation = "/theme/custom/global/";
+const std::string kLocalCustomThemeLocation = "/theme/custom/local/";
+
 // A map from the name of the theme to the location of the file and a boolean representing
 // whether or not the theme is dark.
-typedef std::map<std::string, std::pair<std::string, bool> > ThemeMap;
-void getThemesInLocation(const rstudio::core::FilePath& location, ThemeMap& themeMap)
+typedef std::map<std::string, std::tuple<std::string, std::string, bool> > ThemeMap;
+
+/// @brief Determines whether a string can be evaluated to logical "true". A return value of false
+///        does not indicate that the string can be evaluted to logical "false".
+///
+/// @param toConvert    The string to convert to boolean.
+///
+/// @return True if the string matches 1, TRUE, true, or any case variant of true; false otherwise.
+bool strIsTrue(const std::string& toConvert)
+{
+   return boost::regex_match(
+            toConvert,
+            boost::regex("(?:1|t(?:rue)?)", boost::regex::icase));
+}
+
+/// @brief Determines whether a string can be evaluated to logical "false". A return value of true
+///        does not indicate that the string can be evaluted to logical "true".
+///
+/// @param toConvert    The string to convert to boolean.
+///
+/// @return True if the string matches 0, FALSE, false, or any case variant of false; true
+///         otherwise.
+bool strIsFalse(const std::string& toConvert)
+{
+   return boost::regex_match(
+            toConvert,
+            boost::regex("(?:0|f(?:alse)?)", boost::regex::icase));
+}
+
+/**
+ * @brief Gets themes in the specified location.
+ *
+ * @param location         The location in which to look for themes.
+ * @param themeMap         The map which will contain all found themes after the call.
+ * @param urlPrefix        The URL prefix for the theme. Must end with "/"
+ */
+void getThemesInLocation(
+      const rstudio::core::FilePath& location,
+      ThemeMap& themeMap,
```

Comment:
I don't mind putting them at the end, but using references instead of pointers is preferable whenever there is no reason for the parameter to be null because you can rely on the compiler to prevent programmer-error bugs and avoid null checking.

Reply:
The main reason we use raw pointers is that it's obvious *at the call site* that the object is going to be mutated. From the coding standards:

> Mutable arguments should be passed as dumb pointer (implying external ownership)

Can't disagree that references are safer, however! We can certainly have a conversation about whether we should update the standards, but until then let's stick with them.